src/opcodes/tests/test_opcodes.cairo

```diff
@@ -691,3 +691,21 @@ fn test_op_lessthan_equal() {
     let expected_stack = array![""];
     assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
 }
+
+#[test]
+fn test_op_1negate() {
+    let program = "OP_1NEGATE";
+    let mut compiler = CompilerTraitImpl::new();
+    let bytecode = compiler.compile(program);
+    let mut engine = EngineTraitImpl::new(bytecode);
+    // let _ = engine.step();
+    let res = engine.step();
+    assert!(res, "Execution of run failed");
+    let dstack = engine.get_dstack();
+
+    assert_eq!(dstack.len(), 1, "Stack length is not 1");
+
+    let expected_stack = array!["\x81"];
```

Comment:
Hello, maybe you can try to construct the `expected_stack` like this :
```
let mut expected_bytes: ByteArray = Default::default();
expected_bytes.append_byte(129);
let mut expected_stack = ArrayTrait::<ByteArray>::new();
expected_stack.append(expected_bytes);
assert_eq!(dstack, expected_stack.span(), "Stack is not equal to expected");
```

Reply:
This is way better sorry `let expected_stack = array![129];`

Reply:
There is a utility function for this, you can use `utils::int_to_bytes(-1)`, so something like `let expected_stack = array![utils::int_to_bytes(-1)]`